src/components/badge/index.js

```diff
@@ -107,12 +112,43 @@ export default class Badge extends BaseComponent {
       width += borderWidth * 2;
       height += borderWidth * 2;
     }
-
     return {width, height};
   }
 
   renderLabel() {
-    const {label} = this.props;
+    const {labelLengthFormater} = this.props;
+    let {label} = this.props;
+
+    switch (labelLengthFormater) {
+      case null:
+      case undefined:
+        break;
+      case 1:
+        if (label > 9) {
```

Comment:
the label is string typed, you can compare it to a number, not sure what we'll be returned here.

Reply:
Done. Added isNaN check